src/main/java/org/web3j/tx/HighThroughputTransactionManager.java

```diff
@@ -0,0 +1,95 @@
+package org.web3j.tx;
+
+import org.web3j.crypto.Credentials;
+import org.web3j.crypto.Hash;
+import org.web3j.crypto.TransactionEncoder;
+import org.web3j.protocol.Web3j;
+import org.web3j.protocol.core.DefaultBlockParameterName;
+import org.web3j.protocol.core.methods.request.RawTransaction;
+import org.web3j.protocol.core.methods.response.EthGetTransactionCount;
+import org.web3j.protocol.core.methods.response.EthSendTransaction;
+import org.web3j.protocol.exceptions.TransactionTimeoutException;
+import org.web3j.utils.Numeric;
+
+import java.math.BigInteger;
+import java.util.HashMap;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Created by fernando on 2/1/17.
+ */
+public class HighThroughputTransactionManager extends RawTransactionManager {
+
+	private static HashMap<String,HighThroughputTransactionManager> map = new HashMap<>();
+	private String accountAddress;
+	private BigInteger nonce = BigInteger.valueOf(-1);
+	private HashMap<BigInteger, Object> transactions;
+	private Web3j web3j;
+	private byte chainId;
+
+	public static HighThroughputTransactionManager getInstance(Web3j web3j, Credentials credentials)
```

Comment:
I was thinking on this topic, if we need to put all this code inside the web3j library, so I will decouple a little bit more.  An external service in the client side will manage connections and queues of transactions and logs processed.